src/client/datascience/telemetry/interpreterPackageTracker.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { inject, injectable } from 'inversify';
+import { IExtensionSingleActivationService } from '../../activation/types';
+import { IPythonInstaller, IPythonExtensionChecker } from '../../api/types';
+import { IVSCodeNotebook } from '../../common/application/types';
+import { InterpreterUri } from '../../common/installer/types';
+import { IExtensions, IDisposableRegistry, Product, IConfigurationService } from '../../common/types';
+import { swallowExceptions } from '../../common/utils/decorators';
+import { isResource, noop } from '../../common/utils/misc';
+import { IInterpreterService } from '../../interpreter/contracts';
+import { isLocalLaunch } from '../jupyter/kernels/helpers';
+import { InterpreterPackages } from './interpreterPackages';
+import { NotebookKernel as VSCNotebookKernel } from '../../../../types/vscode-proposed';
+import { isJupyterKernel } from '../notebook/helpers/helpers';
+
+@injectable()
+export class InterpreterPackageTracker implements IExtensionSingleActivationService {
+    private activeInterpreterTrackedUponActivation?: boolean;
+    constructor(
+        @inject(InterpreterPackages) private readonly packages: InterpreterPackages,
+        @inject(IPythonInstaller) private readonly installer: IPythonInstaller,
+        @inject(IExtensions) private readonly extensions: IExtensions,
+        @inject(IPythonExtensionChecker) private readonly pythonExtensionChecker: IPythonExtensionChecker,
+        @inject(IDisposableRegistry) private readonly disposables: IDisposableRegistry,
+        @inject(IInterpreterService) private readonly interpreterService: IInterpreterService,
+        @inject(IVSCodeNotebook) private readonly notebook: IVSCodeNotebook,
+        @inject(IConfigurationService) private readonly configurationService: IConfigurationService
+    ) {}
+    public async activate(): Promise<void> {
+        if (!isLocalLaunch(this.configurationService)) {
+            return;
+        }
+        this.notebook.onDidChangeActiveNotebookKernel(this.onDidChangeActiveNotebookKernel, this, this.disposables);
+        this.interpreterService.onDidChangeInterpreter(this.trackPackagesOfActiveInterpreter, this, this.disposables);
+        this.installer.onInstalled(this.onDidInstallPackage, this, this.disposables);
+        this.extensions.onDidChange(this.trackUponActivation, this, this.disposables);
+        this.trackUponActivation().catch(noop);
```

Comment:
Fetch the list:
* When a kernel is selected in a VSC Notebook
* When packages are installed
* When interpreter changes
* When extension loads
* When Python extension is installed

Reply:
Perhaps you should put this in a comment?